lib/bot.js

```diff
@@ -137,11 +137,15 @@ class Bot {
     logger.debug('Channel Mapping', channelName, this.channelMapping[channelName]);
     if (ircChannel) {
       const username = author.username;
+      const fromServer = message.server;
+      const userDetails = fromServer.detailsOfUser(author);
+      let nickname = userDetails.nick;
```

Comment:
These two could be changed to:
`const nickname = userDetails.nick || author.username;` I think?